components/pca_transformation/__init__.py

```diff
@@ -0,0 +1,32 @@
+from bokeh.embed import json_item
+import pandas as pd
+from sklearn.decomposition import PCA
+import scanpy as sc
+import numpy as np
+
+from components.gene_count_matrix import anndata_from_path
+from components.bokeh import interactive_circle_plot
+
+def pca_transformation(gene_count_matrix):
+  anndata_input = anndata_from_path(gene_count_matrix['url'])
+  df = anndata_input.to_df()
+  df = df.transpose()
+
+  # PCA
+  pca = PCA()
+  transformed_pca = pca.fit_transform(df)
+
+  # leiden coloring
+  leiden_df = sc.AnnData(df,dtype=np.float32)
+  sc.pp.pca(leiden_df)
```

Comment:
1. It looks like you do pca twice..? Just remove the sklearn stuff, scanpy is doing pca anyway, df is already anndata before to convert it to pandas df (remove that) the result of both PCA and leidan are stored in the AnnData object (probably it's df.obs['pca']).
2. Leiden is not a part of PCA, you should change the name (or description) of this step if you're doing both.